components/service-operator/apis/database/v1beta1/postgres_types.go

```diff
@@ -16,60 +16,213 @@ limitations under the License.
+	AWS PostgresAWSSpec `json:"aws,omitempty"`
 	// Secret name to be used for storing relevant instance secrets for further use.
 	Secret string `json:"secret,omitempty"`
 }
 
-// PostgresStatus defines the observed state of Postgres
-type PostgresStatus struct {
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// ID of an instance for a reference.
-	ID string `json:"id"`
-	// Status of the currently running instance.
-	Status string `json:"status"`
-	// Reason for the current status of the instance.
-	Reason string `json:"reason,omitempty"`
-	// Events will hold more in-depth details of the current state of the instance.
-	Events []Event `json:"events,omitempty"`
-}
-
 // +kubebuilder:object:root=true
 
 // Postgres is the Schema for the postgres API
 type Postgres struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+	object.Status     `json:"status,omitempty"`
+
+	Spec PostgresSpec `json:"spec,omitempty"`
+}
+
+// GetSecretName returns the name of a secret that will get populated with
+// connection details
+func (p *Postgres) GetSecretName() string {
+	if p.Spec.Secret == "" {
+		return p.GetName()
+	}
+	return p.Spec.Secret
+}
+
+// Name returns the name of the SQS cloudformation stack
+func (p *Postgres) GetStackName() string {
+	return fmt.Sprintf("%s-%s-%s-%s", env.ClusterName(), "postgres", p.GetNamespace(), p.GetName())
+}
 
-	Spec   PostgresSpec   `json:"spec,omitempty"`
-	Status PostgresStatus `json:"status,omitempty"`
+// GetStackTemplate implements cloudformation.Stack to serialize the configuration as a cloudformaiton template
+func (p *Postgres) GetStackTemplate() *cloudformation.Template {
+	template := cloudformation.NewTemplate()
+
+	template.Parameters[VPCSecurityGroupIDParameterName] = map[string]interface{}{
+		"Type": "String",
+	}
+	template.Parameters[DBSubnetGroupNameParameterName] = map[string]interface{}{
+		"Type": "String",
+	}
+
+	tags := []cloudformation.Tag{
+		{
+			Key:   "Cluster",
+			Value: env.ClusterName(),
+		},
+		{
+			Key:   "Service",
+			Value: "sqs",
+		},
+		{
+			Key:   "Name",
+			Value: p.GetName(),
+		},
+		{
+			Key:   "Namespace",
+			Value: p.GetNamespace(),
+		},
+		{
+			Key:   "Environment",
+			Value: p.GetNamespace(),
+		},
+	}
+
+	// generate secret in cloudformation not in operator (keeps state in aws)
+	template.Resources[PostgresResourceMasterCredentials] = &cloudformation.AWSSecretsManagerSecret{
+		Description: "Master Credentials for postgres instance",
+		GenerateSecretString: &cloudformation.GenerateSecretString{
+			SecretStringTemplate: `{"username": "master"}`,
+			ExcludeCharacters:    `"@/\'`,
+			GenerateStringKey:    "password",
+			PasswordLength:       32,
+		},
+	}
+
+	// create a reference to the values created in secrets manager
+	masterUsernameSecretRef := cloudformation.Join("", []string{"{{resolve:secretsmanager:", cloudformation.Ref(PostgresResourceMasterCredentials), ":SecretString:username}}"})
```

Comment:
Expecting this SecretsManager resolve thing to work in a CloudFormation Join isn't something I'd necessarily expect to work, but if it does work, ok.